### ISSUE TITLE
[Crash + Some questions] Recipe editing.

### ISSUE THREAD

**Issue by OneEyeMaker** [ReikaKalseki/Reika_Mods_Issues#438]
Hello.
1) After enabling `"Allow Nonstandard Recipe Modifications"` and `"Strong Recipe Editing"` in config of RotaryCraft and relaunching Minecraft, I got this crash:
http://pastebin.com/ngaMmh8i
It seems, that you confused `=` and `:` [here](https://github.com/ReikaKalseki/DragonAPI/blob/master/Instantiable/IO/ControlledConfig.java#L389), because in `RotaryCraft_RecipeModding.cfg` I have line separated with `=`.

2) As I understand, `Strong Recipe Editing` provides some advanced configuration options. But which options does this file provide? I don't find any information abount that. Is it WIP? Please, explain...

3) I know, that you don't like any recipe editors (you prefer config options instead). But _you can't imagine **all** ideas of modpack creators_ => _you can't provide config options for all these ideas_ => _you should provide (some) freedom of creativity for modpack makers_.
In other words, add config option to disable [this functional](https://github.com/ReikaKalseki/DragonAPI/blob/master/ModInteract/DeepInteract/MTInteractionManager.java)  :))

P. S. Sorry for my English, I'm from Russia.

**Comment by ReikaKalseki** [comment]
The crash appears to be an error in the config handling system, which you may have identified.

The recipe editing system would allow for things like MT to work. However, I need code on his end to be implemented as well for such a thing to be possible.

**Comment by OneEyeMaker** [comment]
> However, I need code on his end to be implemented as well for such a thing to be possible.

AFAIK, the most of the code you can implement at your side by using `@ZenClass`, `@ZenMethod` (when it comes to MT).

**Comment by ReikaKalseki** [comment]
I tried that, and it never worked:

https://github.com/ReikaKalseki/RotaryCraft/pull/26

**Comment by OneEyeMaker** [comment]
Firstly, please, listen me and _my_ vision of this situation; after that _I'll listen you_.
1) If you want to add MT integration, you should allow to edit your standard (crafting/smelting/OreDict registration) recipes. I don't speak about allowing to edit _**all**_ recipes; but most of recipes should be editable, if `RECIPEMOD` and/or `STRONGRECIPEMOD` are enabled. 

In other words, you should disable [this](https://github.com/ReikaKalseki/DragonAPI/blob/master/DragonAPIInit.java#L533), if `RECIPEMOD` and/or `STRONGRECIPEMOD` are enabled in config.

2) Some things about your MT handlers. See [here](https://github.com/ReikaKalseki/DragonAPI/blob/master/DragonAPIInit.java#L506) and [here](https://github.com/ReikaKalseki/RotaryCraft/blob/master/RotaryCraft.java#L472).

Why do you create class just for storing handlers in `HashSet`, if you never use this set? You can do that much simplier:

```
//Pseudo-code for DragonAPI (post-)init.
if (isMTLoaded)
{
    if (isRoCLoaded)
    {
        MineTweakerAPI.registerClass(GrinderTweaks.class);
        MineTweakerAPI.registerClass(PulseJetTweaks.class);
        //...
        MineTweakerAPI.registerClass(AnotherMachineTweaks.class);
    }
    if (isReCLoaded)
    {
        //...
    }
    //...
}
```

_More native = more effective_.

3).[GrinderTweaks](https://github.com/ReikaKalseki/RotaryCraft/blob/master/ModInterface/Minetweaker/GrinderTweaker.java)

There are few problems:
    a) This class may be not registered properly (because you have registration code in _different_ mods, see 2 above).
    b) You never use `output` [here](https://github.com/ReikaKalseki/RotaryCraft/blob/master/ModInterface/Minetweaker/GrinderTweaker.java#L54) => you always get NPE [here](https://github.com/ReikaKalseki/RotaryCraft/blob/master/ModInterface/Minetweaker/GrinderTweaker.java#L67), because `output` always == null. (That NPE doesn't cause crash, because MT has `try...catch`  at some point).
    c) Use qualified name [here](https://github.com/ReikaKalseki/RotaryCraft/blob/master/ModInterface/Minetweaker/GrinderTweaker.java#L67) => `GrinderTweaks.grinder` instead of `grinder` (implicit call of property of outer class makes code unreadable).

[PulseJetTweaks](https://github.com/ReikaKalseki/RotaryCraft/blob/master/ModInterface/Minetweaker/PulseJetTweaker.java) and [FrictionTweaks](https://github.com/ReikaKalseki/RotaryCraft/blob/master/ModInterface/Minetweaker/FrictionTweaker.java) have the same problems.

I'm pleased to know _your vision_ of this situation. And I'm ready to help you _anyway_. 
**Thank you for your work.**

P. S. And sorry for my English...

**Comment by ReikaKalseki** [comment]
Your English is not a problem. Your (lack of) understanding of how any of my code (or perhaps Java in general) works or is designed, especially when coupled with (what appears to be) an aggressive and self-righteous attitude, however, is.

3b), Actual issue or not, is _not_ the cause of the problem I am having, which is that **Minetweaker cannot even see the classes I register**. So any error in handing a command, though it may be a second error, is not even being loaded, much less causing said issue.

3c) is you complaining about formatting. Not a bug, much less something that will break anything.

1) is irrelevant to the problem I am having. Yes, I will need to give it exceptions (not disable it entirely for everything as you state is required). But that again is not what is making MT not even see my classes.

2): What are you talking about, "never use the [HashSet]"? It is explicitly, obviously used halfway down the class, when I iterate over it.

As for your pseudocode, DragonAPI is a core library. It does not manage other mods. Your version would not even compile, because it is making hard reference to classes in other mods.

That takes me to 3a.

The way this handling is implemented is the way that all of the mods and intermod handling work, and how they will always work. DragonAPI only registers the core accesses and manages the actual registration. Other mods are left to register their individual handlers to DragonAPI's manager.

I do not find it at all likely that this would break Minetweaker handling - it changes nothing at all about how the method call works, and you could not even _detect_ what mod did the registration without going far out of your way to do so. In the unlikely chance that you are correct, then the error is on his end, because that would be a ridiculous and unreasonable restriction that is fundamentally incompatible with my mod structure, a structure which I am not going to violate for this.

**Comment by TomeWyrm** [comment]
Tiny nitpick. That "You English" at the beginning of your comment should be "Your English" Reika.

**Comment by ReikaKalseki** [comment]
Fixed.

**Comment by ReikaKalseki** [comment]
This is not necessary anymore, given my `LuaBlock` system.
